node/include/cocaine/detail/service/node/slave/machine.hpp

```diff
@@ -108,12 +109,39 @@ class machine_t : public std::enable_shared_from_this<machine_t> {
 
         metrics::shared_metric<metrics::gauge<std::string>> state;
         metrics::shared_metric<metrics::gauge<std::uint64_t>> uptime;
+        metrics::shared_metric<metrics::gauge<double>> concurrency;
 
         metrics_t(context_t& context, std::shared_ptr<machine_t> parent);
     };
 
     std::unique_ptr<metrics_t> metrics;
 
+    typedef metrics::usts::ewma_t ewma_type;
+
+    struct metrics_data_t {
```

Comment:
To make single "gather point" for  _machine_t_ metrics (which could be added in future). We can use _metrics_t_ as such point, but afaik it is design as a kind of watcher/metrics registrant, not to hold metrics state, also it seems to have lifetime not to span along with _machine_t_  lifetime.